File: src/Python/849.MaxDistToClosest.py

class Solution:
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        count = 0
        maxcount = 0
        for i, seat in enumerate(seats):
            if seat == 1:
                if count > maxcount:
                    maxcount = count
                count = 0
            else:
                count += 1
        return max((maxcount + 1) // 2, count, seats.index(1))

if __name__ == '__main__':
    solution = Solution()
    print(solution.maxDistToClosest([1,0,0,0,1,0,1]))
    print(solution.maxDistToClosest([1,0,0,0]))
    print(solution.maxDistToClosest([0,0,1]))
else:
    pass

File: src/Java/849.MaxDistToClosest.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Set;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int maxDistToClosest(int[] seats) {
        int count = 0;
        int maxcount = 0;
        int firstindex = 0;   
        for(int i = 0; i < seats.length; i++){
            if (seats[i] == 1){
                firstindex = i;
                break;
            }    
        }
        for(int i = 0; i < seats.length; i++){
            if (seats[i] == 1){
                if (count > maxcount)
                    maxcount = count;
                count = 0;
            }
            else
                ++count;
        }
        return Math.max((maxcount + 1) / 2, Math.max(count, firstindex));
    }
}
